playbooks/group_vars/all.yml

```diff
@@ -0,0 +1,49 @@
+---
```

Comment:
While I agree a group_var is very convenient to use, I don't see a reason to load that on all nodes.

Likewise, there seem to be a big overlap with common-vars.

Could you edit common-vars instead, focusing on the existing vars to migrate to this new location?

Reply:
For me, explicitly including common-vars wherever needed seems like additional step. It may not be obvious to someone who is new to code. Also common-vars are intended to be shared across all roles and in a way indirectly shared across all nodes.

I have commented out common_vars and may be we can remove the file later if group_vars seems to work for us.

Reply:
I don't think it's a good idea.

These group_vars would apply to all the node, therefore overloading the ansible worker of data that are pretty much useless. Instead, loading the var at runtime would be easier to scale.

The idea of common-vars is indeed to load them from the roles. It is, according to me, simple to understand: When you read the role, it explicitly tells to load the vars.

With group vars + load vars, it becomes less obvious.

Reply:
Online, when I looked for pattern around this problem, came across many recommendation of using group_vars for sharing data across roles based on group of hosts.

In general, ansible already provides this mechanism to define variables and to propagate them across group of nodes or all nodes. For simplicity and with current state of things (conflicting host group_names), just added to all. 
We can certainly add vars to group_vars/osh-workers if those are only intended for osh worker nodes during osh mode of deployment.

common-vars.yml is not a obvious place for looking variables for a newcomer as its not a standard practice and then require extra step of including it in specific roles before you can use it (came across issue myself as assumed common-vars is already included somewhere but it was not).